src/server/snapshot.cc

```diff
@@ -115,6 +115,9 @@ void SliceSnapshot::FinalizeJournalStream(bool cancel) {
   VLOG(1) << "FinalizeJournalStream";
   DCHECK(db_slice_->shard_owner()->IsMyThread());
   if (!journal_cb_id_) {  // Finalize only once.
```

Comment:
worth adding a small test then ?

p.s. haven't checked the actual issue/solution -- will do this on Monday